src/storage/store/var_list_column.cpp

```diff
@@ -145,19 +182,26 @@ void VarListColumn::scanFiltered(Transaction* transaction, node_group_idx_t node
         auto pos = resultVector->state->selVector->selectedPositions[i];
         auto startOffsetInStorageToScan = listOffsetInfoInStorage.getListOffset(pos);
         auto endOffsetInStorageToScan = listOffsetInfoInStorage.getListOffset(pos + 1);
+        auto dataVector = ListVector::getDataVector(resultVector);
+        if (!dataVector->state) {
+            dataVector->setState(std::make_unique<DataChunkState>());
```

Comment:
Double check here to see when `dataVector->state` is nullptr.
We might have set state for dataVector somewhere else. better understand this better before we add an ad-hoc `if` check here.